arbitrum-docs/launch-orbit-chain/how-tos/customize-arbos.mdx

```diff
@@ -0,0 +1,148 @@
+        )
+        panic(message)
+    }
+}
+nextArbosVersion := state.arbosVersion + 1
+switch nextArbosVersion {
+    case 1:
+        //.....
+    case 2:
+        //....
+    //.....
+    case 21:
+        // Set your new ArbOS state value here
+        ensure(state.SetNewMyNumber(${random number}))
+    //....
+} 
+```
+
+Here, we will ensure that the initial value of $(a random number) after ArbOS is upgraded to V21.
+
+
+:::caution
+It should be noted that when you initialize the state (initial code is in [customize precompile Option 5](./customize-precompile.mdx#option-5-call-and-modify-state)), you need to initialize it to 0 or null value first to avoid a potential re-org on your chain.
+:::
+
+Also, please make sure your program cannot call the `state.SetNewMyNumber` or other functions may change the value of `myNumber` before ArbOS V21. 
+To prevent this, if you are using an external call to the precompile contract to change the value, you can refer to [point 1](./customize-arbos.mdx#1-add-a-new-method-to-exsiting-precompile-on-a-specific-arbos-version) or
+[point 2](./customize-arbos.mdx#2-create-a-new-precompile-contract-on-a-specific-arbos-version) to set the activation time of the precompile contract method.
+If your nitro code needs to call this method to change the state, you can continue reading [point 4](./customize-arbos.mdx#4-any-changes-in-the-stf-logic-that-will-affect-the-final-execution-result).
+
+### 4. Any changes in the STF logic that will affect the final execution result
+If you change the logic in STF, it will cause the execution result of the transaction to be different, 
+you need to keep the original execution logic and put the new logic into another branch.
+You can use an `if else` statement to control it.
+
+For example, we change the `SayHi` return after upgrading the ArbOS version:
+```go
+// The method in ArbHi precompile
+func (con *ArbHi) SayHi(c ctx, evm mech) (string, error) {
+    if p.state.ArbOSVersion() >= ${The arbos you want to upgrade to} {
+        // Your new logic code
+        return "hi, new ArbOS version", nil
+    } else {
+        // The old logic code needs to be kept
+        return "hi", nil
+    }
+}
+```
+
+In the above code, we use precompiles as an example. Some logic might also affect the STF, such as the methods in [block_process.go][block_processor_file_link], [internal_tx.go][internal_tx_file_link], [tx_processor.go][tx_processor_file_link] and so on.
+You also need to use those ArbOS control methods for different logic versions.
+
+:::tip Backward compatibility
+
+Wasm module roots are backward compatible, so upgrading them before an ArbOS version upgrade will not disrupt your chain's functionality.
+
+:::
+
+## Upgrade WASM Module Root on the parent chain
+Afterward, since Arbitrum might need to execute the validation and fraud proofs on the parent chain, you need to update the Wasm Module Root recorded on the parent chain.
```

Comment:
```suggestion
After you have made your custom ArbOS changes, you will need to update the WASM Module root recorded on the parent chain. This is because an Arbitrum chain may execute the validation and fraud proofs on the parent chain.
```

What does it mean to "execute validation on the parent chain"?